### ISSUE TITLE
Error in Shadow Dom polyfill when name an integer

### ISSUE THREAD

**Issue by chimon2000** [webcomponents/webcomponentsjs#256]
In the Shadow Dom polyfills, on the rare occasion that the `getGetter` and `getSetter` methods get a name parameter that is an integer, the following error is thrown:

`Error: Expected ';'`

Changing 
````
return hasEval && isIdentifierName(name) ? new Function("return this.__impl4cf1e782hg__." + name) : function() {
        return this.__impl4cf1e782hg__[name];
};
````

to 
````
return hasEval && isIdentifierName(name) ? new Function("return this.__impl4cf1e782hg__['" + name + "']") : function () {
        return this.__impl4cf1e782hg__[name];
};
````

and 
````
return hasEval && isIdentifierName(name) ? new Function("v", "this.__impl4cf1e782hg__." + name + " = v") : function(v) {
        this.__impl4cf1e782hg__[name] = v;
};
````

to 
````
return hasEval && isIdentifierName(name) ? new Function("v", "this.__impl4cf1e782hg__['" + name + "']" + " = v") : function (v) {
        this.__impl4cf1e782hg__[name] = v;
};
````

resolves this issue. I would assume the same issue would occur for the `getMethod`

**Comment by jmesserly** [comment]
unfortunately, that fix would remove the performance optimization.
I think a better fix would be to isIdentifierName (in src/ShadowDOM/wrappers.js):
```js
  function isIdentifierName(name) {
    return /^\w[a-zA-Z_0-9]*$/.test(name);
  }
```
something like:
```js
  function isIdentifierName(name) {
    return /^\w[a-zA-Z_$][a-zA-Z_$0-9]*$/.test(name);
  }
```

just curious, how did you reproduce the problem of an integer name? I'm trying to think of a DOM API that works like that.

**Comment by chimon2000** [comment]
I'm developing an [app for Office](https://msdn.microsoft.com/en-us/library/office/jj220082.aspx), which has shoddy HTML5 support at best. While it isn't explicitly stated what browser context these apps run under they list having IE9 as a [requirement](https://msdn.microsoft.com/EN-US/library/office/dn833104.aspx).  

Since sometimes it loads fine and other times I get that error w/ epic failure, I'm assuming it's either using 9 or 10 depending on how it feels.

The only other way I was able to reproduce it in a browser was to load my app in IE and switch from Edge to 9.  I know that the new documentation states polymer only supports  >= IE10.

I swapped out your sweet fix for mine and that looks to have resolved the issue (both in browser and in office) as well.
